test/NuGet.Core.Tests/NuGet.Configuration.Test/PackageSourceProviderTests.cs

```diff
@@ -822,6 +822,56 @@ public void LoadPackageSources_ReadsSourcesWithProtocolVersionFromPackageSourceS
                     });
         }
 
+        [Theory]
+        [InlineData(true, null, null)]
+        [InlineData(true, "3", null)]
+        [InlineData(true, "3", "false")]
+        [InlineData(true, "2", "true")]
+        [InlineData(false, null, "true")]
+        [InlineData(false, "", "true")]
+        public void LoadPackageSources_ReadsSourcesWithAllowInsecureConnectionsFromPackageSourceSections(bool useStaticMethod, string protocolVersion, string allowInsecureConnections)
```

Comment:
I know people like keeping code to a minimum, but I would say this is a good time to create multiple tests. When there's significant branching in Assertions, that makes it hard to understand what it means when the test fails. Also, the name of the method doesn't indicate the criteria and success conditions.

eg, we could have (just example wordings):

`LoadPackageSources_WithNullProtocolVersion_LoadsDefault`
`LoadPackageSources_WithNullAllowInsecureConnections_LoadsDefault`
...
`LoadPackageSources_WithNotNullProtocolVersion_LoadsValue`

Reply:
+1 to above comment.

Reply:
Thanks! Fixed.